dask/tests/test_base.py

```diff
@@ -885,3 +885,23 @@ def f(x):
     workers = {i.worker_id for i in prof.results}
 
     assert len(workers) == num_workers
+
+
+@pytest.mark.parametrize('scheduler', ['threads', 'processes'])
+def test_num_workers_env_var(monkeypatch, scheduler):
+    num_workers = 3
+    monkeypatch.setenv('DASK_NUM_WORKERS', num_workers)
+    dask.config.refresh()
+
+    @delayed
+    def f(x):
+        time.sleep(0.5)
+        return x
+
+    a = [f(i) for i in range(5)]
+    with Profiler() as prof:
+        a = compute(*a, scheduler=scheduler)
+
+    workers = {i.worker_id for i in prof.results}
+
+    assert len(workers) == num_workers
```

Comment:
I recommend that we test this functionality more directly.  For example we might set an environment variable that you ahve here, and then verify that the correct value is returned by both `dask.config.get('num_workers')` and dask.config.get('num-workers')`.  We might also use some more generic key, like `a-b` / `a_b`.  It would also be nice to verify that you can use either one as we roundtrip between get/set rather than using environment variables.  

I think that a more direct test will also have the advantage of being faster (this one takes a second or more if I recall correctly) and will also probably communicate the intent of these changes to future developers more clearly.